src/engines/gnucap.c

```diff
@@ -60,6 +60,9 @@ static struct analysis_tag analysis_tags[] = {
 #define TAGS_COUNT (sizeof(analysis_tags) / sizeof(struct analysis_tag))
 
 // Parser STATUS
+typedef enum { STATE_IDLE, IN_VARIABLES, IN_VALUES, STATE_ABORT } ParseDataState;
+static ParseDataState state;
+
```

Comment:
It fixes random parser failures, without introducing side effects such as the one that you mention.

In my opinion, you can safely apply the above mentioned change and that would greatly improve the gnucap engine.